packages/client/hmi-client/src/services/model-configurations.ts

```diff
@@ -249,6 +265,63 @@ export function setParameterSource(
 	}
 }
 
+export function getParameterConstant(config: ModelConfiguration, parameterId: string): number {
+	const parameter = getParameter(config, parameterId);
+	return parameter?.value ?? NaN;
+}
+
+export function setParameterConstant(
+	config: ModelConfiguration,
+	parameterId: string,
+	value: string
+): void {
+	const parameter = getParameter(config, parameterId);
+	if (parameter) {
+		parameter.value = parseFloat(value);
+	}
+}
+
+// sets the parameter distribution type along with its default respective parameters
+export function setParameterDistributionType(
+	config: ModelConfiguration,
+	parameterId: string,
+	distributionType: DistributionType
+): void {
+	const parameter = getParameter(config, parameterId);
+	if (!parameter) return;
+	switch (distributionType) {
+		case DistributionType.Constant:
+			delete parameter.distribution;
+			break;
+		case DistributionType.Uniform:
+			parameter.distribution = {
+				type: DistributionType.Uniform,
+				parameters: {
+					minimum: 0,
+					maximum: 0
+				}
+			};
+			break;
+		default:
+			break;
+	}
+}
+
+// sets the parameter distribution parameters (i.e. minimum and maximum for uniform distribution)
+export function setParameterDistributionParameters(
+	config: ModelConfiguration,
+	parameterId: string,
+	parameters: { [index: string]: any }
+): void {
+	const parameter = getParameter(config, parameterId);
+	if (parameter && parameter.distribution) {
+		parameter.distribution.parameters = {
+			...parameter.distribution.parameters,
+			...parameters
+		};
+	}
```

Comment:
Not sure what the intention is here, but this doesn't look like it works, as you are doing a disjunctive join. For example:

```
p = { mean: 0, std: 0.2 }
p2 = { max: 10, min: 3 }
p = { ...p, ...p2 }
```

We get for p = `{mean: 0, std: 0.2, max: 10, min: 3}` which does not properly describe anything...

Reply:
It works in its current state 😄, but its definitely hacky. Ill rethink this implementation

Reply:
i've might a slight adjustment, the function will now update the keys that currently exist in the distribution object from the given parameters array. i.e

```
p = { min: 0, max: 2}
p2 = { min: 1, std: 8 }
```

Now p will = `{ min: 1, max: 2 }`

This function will be strictly for updating the distribution parameters that are already given in the initialized distribution and not add additional parameters that are not needed.